src/gwt/src/org/rstudio/core/client/widget/ModalDialogBase.java

```diff
@@ -718,16 +713,53 @@ public void refreshFocusableElements()
             "[tabindex=\"0\"]:not([hidden]):not([disabled]), summary:not([hidden]), " +
             "[contenteditable]:not([hidden]), audio[controls]:not([hidden]), " +
             "video[controls]:not([hidden])";
-      NodeList<Element> focusable = DomUtils.querySelectorAll(getElement(), focusableElements);
-      if (focusable.getLength() == 0)
+      NodeList<Element> potentiallyFocusable = DomUtils.querySelectorAll(getElement(), focusableElements);
+
+      ArrayList<Element> focusable = new ArrayList<>();
+      for (int i = 0; i < potentiallyFocusable.getLength(); i++)
+      {
+         // only include items taking up space
+         if (potentiallyFocusable.getItem(i).getOffsetWidth() > 0 && 
+               potentiallyFocusable.getItem(i).getOffsetHeight() > 0)
+         {
+            focusable.add(potentiallyFocusable.getItem(i));
+         }
+      }
+      return focusable;
+   }
```

Comment:
Years away from element queries being implementable in our app probably but they'd sure be nice here.